master/internal/api_checkpoint.go

```diff
@@ -73,16 +73,26 @@ func (m *Master) canDoActionOnCheckpoint(
 	return nil
 }
 
+/* func (m *Master) canDoActionOnCheckpointThroughModel(ctx context.Context, curUser model.User,
+	id string) error {
+
+	modelauth.AuthZProvider.Get().CanGetModel
+	return nil
+}. */
 func (a *apiServer) GetCheckpoint(
 	ctx context.Context, req *apiv1.GetCheckpointRequest,
 ) (*apiv1.GetCheckpointResponse, error) {
 	curUser, _, err := grpcutil.GetUser(ctx)
 	if err != nil {
 		return nil, err
 	}
-	if err = a.m.canDoActionOnCheckpoint(ctx, *curUser, req.CheckpointUuid,
-		expauth.AuthZProvider.Get().CanGetExperimentArtifacts); err != nil {
-		return nil, err
+	errE := a.m.canDoActionOnCheckpoint(ctx, *curUser, req.CheckpointUuid,
+		expauth.AuthZProvider.Get().CanGetExperimentArtifacts)
+	// Nikita TODO: get model from checkpointUuid and then use CanGetModel
```

Comment:
TODO: each checkpoint can be associated with multiple model ids. @ioga @maxrussell @stoksc so if the user has permissions in any of the models' workspace then it's okay to get the checkpoint?

Reply:
[Confirmed with Haran](https://determined-ai.slack.com/archives/C03LE5Z8R46/p1673988924756529?thread_ts=1673988356.521749&cid=C03LE5Z8R46) that this makes sense. working on it!